frontend/src/v5/ui/routes/viewer/viewer.tsx

```diff
@@ -15,23 +15,115 @@
+import { PROJECTS_LIST_ROUTE, ViewerParams } from '../routes.constants';
 import { useFederationsData } from '../dashboard/projects/federations/federations.hooks';
+import { useContainersData } from '../dashboard/projects/containers/containers.hooks';
+import { NoRevisionOverlay } from './noRevisionOverlay';
 
 export const Viewer = () => {
 	const { teamspace, containerOrFederation, revision } = useParams<ViewerParams>();
+	const history = useHistory();
 	const v4Match = {
 		params: {
 			model: containerOrFederation,
 			teamspace,
 			revision,
 		} };
 
-	useFederationsData();
-	useContainersData();
+	const { federations } = useFederationsData();
+	const { containers } = useContainersData();
+
+	const getContainerFromId = (containerId: string) => containers.find((container) => container._id === containerId);
+	const getFederationFromId = (federationId: string) => federations.find(
+		(federation) => federation._id === federationId,
+	);
+	const selectedFederation = getFederationFromId(containerOrFederation);
+	const selectedContainer = getContainerFromId(containerOrFederation);
+
+	const checkLatestRevisionReady = (container) => {
+		if ((!canUploadToBackend(container.status))
+			&& container.revisionsCount
+		) {
+			DialogsActionsDispatchers.open('info', {
+				title: formatMessage(
+					{ id: 'viewer.latestRevisionNotReady.title', defaultMessage: 'The latest revision is still processing' },
+				),
+				message: formatMessage({
+					id: 'viewer.latestRevisionNotReady.message',
+					defaultMessage: 'Until processing has completed, we can only show the latest available revision.',
+				}),
+				primaryButtonLabel: formatMessage({
+					id: 'viewer.latestRevisionNotReady.primaryLabel',
+					defaultMessage: 'Go to viewer',
+				}),
+				onClickSecondary: () => {
+					history.push(generatePath(PROJECTS_LIST_ROUTE, { teamspace }));
+				},
+			});
+		}
+	};
+
+	useEffect(() => {
+		if (selectedContainer) {
+			checkLatestRevisionReady(selectedContainer);
+		}
+		if (selectedFederation) {
+			selectedFederation.containers.forEach(
+				(containerId) => checkLatestRevisionReady(getContainerFromId(containerId)),
+			);
+		}
+	}, [selectedContainer, selectedFederation]);
+
+	if (selectedContainer && !selectedContainer.hasStatsPending) {
+		if (!selectedContainer.revisionsCount) {
+			const message = canUploadToBackend(selectedContainer.status) ? (
+				formatMessage({
+					id: 'noRevisionOverlay.subheading.container.notProcessing',
+					defaultMessage: 'You\'ll need to upload a new revision.',
+				})
+			) : (
+				formatMessage({
+					id: 'noRevisionOverlay.subheading.container.processing',
+					defaultMessage: 'The Container is empty, you\'ll need to wait for the Container to finish processing.',
+				})
+			);
+			return (
+				<NoRevisionOverlay
+					isContainer
+					message={message}
+				/>
+			);
+		}
+	}
+
+	if (selectedFederation && !selectedFederation.hasStatsPending) {
```

Comment:
Maybe this could be change to this:
```suggestion
	if (!selectedFederation?.hasStatsPending) {
```